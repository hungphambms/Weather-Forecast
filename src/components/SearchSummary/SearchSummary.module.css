.container {
  margin: 40px 0 0;
  position: relative;
}

.container h3 {
  font-size: 18px;
  padding: 0;
  margin: 0 0 8px;
  color: var(--color-dark);
}

.container p {
  font-size: 16px;
  color: var(--color-medium);
}

.container label {
  position: absolute;
  top: calc(50% - 24px);
  right: 24px;

  font-size: 32px;
  color: var(--color-success);
}