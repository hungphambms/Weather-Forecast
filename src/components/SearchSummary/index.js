import React from "react";
import PropTypes from "prop-types";

import styles from "./SearchSummary.module.css";
import { k2c } from "../../utils/temperature";

function SearchSummary({ summary }) {
  if (!summary || !summary.current) {
    return null;
  }

  const { name, timezone, current } = summary;

  return (
    <div aria-label="search-summary" className={styles.container}>
      <h3>{name}</h3>
      <p>
        <strong>Timezone:</strong> {timezone}
      </p>

      <label>
        <img
          src={`http://openweathermap.org/img/w/${current.weather[0].icon}.png`}
          alt="Temp Icon"
        />

        {k2c(current.temp)}
      </label>
    </div>
  );
}

SearchSummary.defaultProps = {
  summary: {},
};

SearchSummary.propTypes = {
  summary: PropTypes.object,
};

export default SearchSummary;
