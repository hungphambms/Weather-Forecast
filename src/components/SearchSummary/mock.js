export const MOCK_DATA = {
  id: 1566083,
  name: "Ho Chi Minh City",

  lat: 10.75,
  lon: 106.6667,
  timezone: "Asia/Ho_Chi_Minh",
  timezone_offset: 25200,
  current: {
    dt: 1615014591,
    sunrise: 1614985544,
    sunset: 1615028631,
    temp: 307.15,
    feels_like: 307.65,
    pressure: 1011,
    humidity: 36,
    dew_point: 289.98,
    uvi: 9.28,
    clouds: 40,
    visibility: 10000,
    wind_speed: 2.57,
    wind_deg: 40,
    weather: [
      {
        id: 802,
        main: "Clouds",
        description: "scattered clouds",
        icon: "03d",
      },
    ],
  }
};
