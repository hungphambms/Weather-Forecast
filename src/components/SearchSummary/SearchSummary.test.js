import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { MOCK_DATA } from "./mock";

import SearchSummary from "./index";

const SEARCH_SUMMARY_PROPS = MOCK_DATA;

describe("SearchSummary rendering", () => {
  it("renders SearchSummary component without crashing", () => {
    const component = renderer.create(<SearchSummary summary={SEARCH_SUMMARY_PROPS} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders SearchSummary component when passed correct props", () => {
    const { getByLabelText, unmount } = render(<SearchSummary summary={SEARCH_SUMMARY_PROPS} />);

    const headingContainer = getByLabelText("search-summary");
    expect(headingContainer.textContent).not.toBeNull();

    unmount();
  });
});
