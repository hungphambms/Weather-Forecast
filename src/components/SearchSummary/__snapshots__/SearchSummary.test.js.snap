// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSummary rendering renders SearchSummary component without crashing 1`] = `
<div
  aria-label="search-summary"
  className="container"
>
  <h3>
    Ho Chi Minh City
  </h3>
  <p>
    <strong>
      Timezone:
    </strong>
     
    Asia/Ho_Chi_Minh
  </p>
  <label>
    <img
      alt="Temp Icon"
      src="http://openweathermap.org/img/w/03d.png"
    />
    34
  </label>
</div>
`;
